src/ucp/core/ucp_ep.c

```diff
@@ -894,29 +896,42 @@ static ucs_status_t ucp_ep_close_nb_check_params(ucp_ep_h ep, unsigned mode)
 }
 
 ucs_status_ptr_t ucp_ep_close_nb(ucp_ep_h ep, unsigned mode)
+{
+    const ucp_request_param_t param = {
+        .op_attr_mask = UCP_EP_PARAM_FIELD_FLAGS,
+        .flags        = (mode == UCP_EP_CLOSE_MODE_FORCE) ?
+                        UCP_EP_CLOSE_FLAG_FORCE : 0
+    };
+
+    return ucp_ep_close_nbx(ep, &param);
+}
+
+ucs_status_ptr_t ucp_ep_close_nbx(ucp_ep_h ep, const ucp_request_param_t *param)
 {
     ucp_worker_h  worker = ep->worker;
+    int           force;
+    uint32_t      flags;
     void          *request;
     ucp_request_t *close_req;
     ucs_status_t  status;
+    unsigned      uct_flags;
 
-    status = ucp_ep_close_nb_check_params(ep, mode);
+    flags  = ucp_request_param_flags(param);
+    status = ucp_ep_close_nbx_check_params(ep, flags, &force);
```

Comment:
maybe not use a separate function ucp_ep_close_nbx_check_params() since it's used only once?

Reply:
or just invoke `ucp_request_param_flags` inside `ucp_ep_close_nbx_check_params` once